python/lsst/pipe/tasks/configurableActions/_configurableActionStructField.py

```diff
@@ -0,0 +1,330 @@
+
+    # create descriptors to handle special update and remove behavior
+    update = ConfigurableActionStructUpdater()
+    remove = ConfigurableActionStructRemover()
+
+    def __init__(self, config: Config, field: ConfigurableActionStructField,
+                 value: Mapping[str, ConfigurableAction], at: Any, label: str):
+        object.__setattr__(self, '_config', config)
+        object.__setattr__(self, '_attrs', {})
+        object.__setattr__(self, '_field', field)
+        object.__setattr__(self, '_history', [])
+
+        self.history.append(("Struct initialized", at, label))
+
+        if value is not None:
+            for k, v in value.items():
+                setattr(self, k, v)
+
+    @property
+    def history(self) -> List[tuple]:
+        return self._history
+
+    @property
+    def fieldNames(self) -> Iterable[str]:
+        return self._attrs.keys()
+
+    def __setattr__(self, attr: str, value: Union[ConfigurableAction, Type[ConfigurableAction]],
+                    at=None, label='setattr', setHistory=False) -> None:
+
+        if hasattr(self._config, '_frozen') and self._config._frozen:
+            msg = "Cannot modify a frozen Config. "\
+                  f"Attempting to set item {attr} to value {value}"
+            raise FieldValidationError(self._field, self._config, msg)
+
+        if attr not in (self.__dict__.keys() | type(self).__dict__.keys()):
+            name = _joinNamePath(self._config._name, self._field.name, attr)
+            if at is None:
+                at = getCallStack()
+            if isinstance(value, ConfigurableAction):
+                valueInst = type(value)(__name=name, __at=at, __label=label, **value._storage)
+            else:
+                valueInst = value(__name=name, __at=at, __label=label)
+            self._attrs[attr] = valueInst
+        else:
+            super().__setattr__(attr, value)
+
+    def __getattr__(self, attr):
+        if attr in object.__getattribute__(self, '_attrs'):
+            return self._attrs[attr]
+        else:
+            super().__getattribute__(attr)
+
+    def __delattr__(self, name):
+        if name in self._attrs:
+            del self._attrs[name]
+        else:
+            super().__delattr__(name)
+
+    def __iter__(self) -> Generator[ConfigurableAction, None, None]:
+        yield from self._attrs.values()
+
+    def items(self) -> Iterable[Tuple[str, ConfigurableAction]]:
+        yield from self._attrs.items()
+
+
+T = TypeVar("T", bound="ConfigurableActionStructField")
+
+
+class ConfigurableActionStructField(Field):
+    """`ConfigurableActionStructField` is a `~lsst.pex.config.Field` subclass
+    that allows `ConfigurableAction` s to be organized in a
+    `~lsst.pex.config.Config` class in a manor similar to how a
+    `~lsst.pipe.base.Struct` works.
+
+    This class implements a `ConfigurableActionStruct` as an intermediary
+    object to organize the `ConfigurableActions`. See it's documentation for
```

Comment:
```suggestion
    object to organize the `ConfigurableAction`\ s. See its documentation for
```

Reply:
I had this in initially but flake8 on github rejected this, saying it was invalid, and the test failed so I removed it. Maybe the repo is miss configured somehow.

Reply:
I think you need to declare the docstring as a raw-string for it to be allowed.